File: C# Fundamentals/C# OOP Basics/11. InterfaceNAbstr/Ex/P09_CollHier/Models/MyList.cs
using System;
using System.Collections.Generic;

public class MyList<T> : AddRemoveCollection<T>, IMyList<T>
{
    private int Used => this.Collection.Count;

    public MyList()
    {
        this.Collection = new List<T>();
    }

    public override T Remove()
    {
        if(this.Collection.Count > 0)
        {
            T element = this.Collection[0];
            this.Collection.RemoveAt(0);
            return element;
        }
        else
        {
            return default(T);
        }
    }

    public override int Add(T element)
    {
        this.Collection.Insert(0, element);
        return 0;
    }
}

File: Java/01.Advanced/01.Advanced/15.Generics/Ex/src/t_07_custom_list/MyList.java
package t_07_custom_list;

import java.util.ArrayList;
import java.util.Collections;
import java.util.List;

public class MyList<T extends Comparable<T>> {
    List<T> collection;

    public MyList() {
        this.collection = new ArrayList<>();
    }

    public void add(T element) {
        collection.add(element);
    }

    public T remove(int index) {
        return collection.remove(index);
    }

    public boolean contains(T element) {
        return collection.contains(element);
    }

    public void swap(int firstIndex, int secondIndex) {
        T temp = collection.get(firstIndex);

        collection.set(firstIndex, collection.get(secondIndex));
        collection.set(secondIndex, temp);
    }

    public int countGreaterThan(T element) {
        int count = 0;

        for (T currElement: collection) {
            if (currElement.compareTo(element) > 0) {
                count++;
            }
        }

        return count;
    }

    public T getMin() {
        return Collections.min(collection);
    }

    public T getMax() {
        return Collections.max(collection);
    }

    public void print() {
        for (T element: collection) {
            System.out.println(element);
        }
    }
}
